sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusRetryPolicy.cs

```diff
@@ -139,22 +140,22 @@ public abstract class ServiceBusRetryPolicy
 
                     catch (Exception ex)
                     {
-                        Exception activeEx = AmqpExceptionHelper.TranslateException(ex);
+                        ExceptionDispatchInfo activeEx = ExceptionDispatchInfo.Capture(AmqpExceptionHelper.TranslateException(ex));
```

Comment:
I'd suggest sliding the `ExceptionDispatchInfo.Capture` to line 158 when you're ready to throw.  The code on lines 144-157 is concerned only with the exception, not the captured context.   Since the `throw` only takes place when not retrying, there's no benefit to having the capture unless you know that you're not retrying.

Reply:
Good suggestion. Will do